### PR TITLE
Refine high availability documentation

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area high-availability
/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR refines the high-availability document WRT the lessons learned when adapting several extensions.

**Special notes for your reviewer**:
/cc @rfranzke 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/development/high-availability.md [hunk 1]
```diff
@@ -11,6 +13,9 @@ spec:
     matchLabels: ...
 ```
 
+2. Capability to run with multiple replicas, e.g. by enabling leader election in case of controller.
+3. Maintain `high-availability-config.resources.gardener.cloud/type={controller, server}` label at `Deployment`s or `StatefulSet`s. If a component is a controller and (webhook) server at the same time, it may be assigned `server`.
```
Comment (rfranzke) [inline docs/development/high-availability.md]: Maybe add a link to [below section](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md#convenient-application-of-these-rules) which explains this part in more detail.

Comment (timuthy) [reply docs/development/high-availability.md]: Right, I missed this section. Maybe we should add all of that information to this section? The actual goal of this PR was to keep everything that developers needs to do (like a TL;DR) in one place.

Comment (rfranzke) [reply docs/development/high-availability.md]: OK, sure

Comment (timuthy) [reply docs/development/high-availability.md]: PTAL 🙂

Comment (rfranzke) [reply docs/development/high-availability.md]: Didn't you also want to move the PDB part into this below section?

Comment (timuthy) [reply docs/development/high-availability.md]: Okay, done. Hope this helps developers to directly get a gist of what needs to be done from their side.

**Review by rfranzke — COMMENTED** [review]
Nice!

/lgtm
/approve

**Review by timebertt — COMMENTED** [review]
Nice, thanks for the enhancement!
/lgtm

**Pull request merged**